bulk_grades/api.py

```diff
@@ -357,12 +367,13 @@ def preprocess_row(self, row):
         Preprocess the CSV row.
         """
         operation = {}
-        if row['user_id'] in self._users_seen:
-            return operation
+        user_id = row['user_id']
+        if user_id in self._users_seen:
+            raise ValidationError(_('Repeated user'))
```

Comment:
`linting`  doesn't allow me to raise a dynamic string. Additionally, error is group by message. So this would produce something like `"Repeat user": [ 1,..5 ]` instead of `"Repeat user aaa": [ 1,...5 ], "Repeat User bbb": [6,...9]`

Reply:
By the way, the line number doesn't tell about the first occurrent because it doesn't trigger the error yet. Let me know if you think it is necessary and I should take care of that.

Reply:
I don't think it's strictly necessary to catch the first row for each user. That said, an AC is to capture the "full set of unique duplicated learners" and this would only capture the first duplicate learner, right?

it's probably worth splitting off a new function, something like `check_for_duplicate_learner_rows` to capture all learners & line numbers and raise a validation error. To @jansenk's point, the messaging should probably be in the view, so the handling of returned error payload should probably live there.

Reply:
To show the first learner, I would need to modify `super-csv` in parallel to this.  `preprocess_row` does not have information about the row number.